sending-a-simple-message-to-a-service/tests/test.bal

```diff
@@ -0,0 +1,69 @@
+import ballerina/http;
+import ballerina/test;
+
+final http:Client cl = check new (string `http://localhost:${port}/healthcare/querydoctor`);
+
+isolated function getSurgeryResponsePayload() returns map<json>[] & readonly => [
+    {
+        "name": "thomas collins",
+        "hospital": "grand oak community hospital",
+        "category": "surgery",
+        "availability": "9.00 a.m - 11.00 a.m",
+        "fee": 7000.0d
+    },
+    {
+        "name": "anne clement",
+        "hospital": "clemency medical center",
+        "category": "surgery",
+        "availability": "8.00 a.m - 10.00 a.m",
+        "fee": 12000.0d
+    },
+    {
+        "name": "seth mears",
+        "hospital": "pine valley community hospital",
+        "category": "surgery",
+        "availability": "3.00 p.m - 5.00 p.m",
+        "fee": 8000.0d
+    }
+];
+
+public client class MockHttpClient {
+    isolated resource function get [string... path](map<string|string[]>? headers = (), http:TargetType targetType = http:Response,
+            *http:QueryParams params) returns http:Response|anydata|http:ClientError {
+        match path[0] {
+            "surgery" => {
+                return getSurgeryResponsePayload();
+            }
+        }
+        return <http:ClientRequestError> error ("unknown specialization", 
+                                            body = string `unknown specialization: ${path[0]}`, 
+                                            headers = {}, 
+                                            statusCode = http:STATUS_NOT_FOUND);
+    }
+}
```

Comment:
An alternative approach for mocking: stubbing. https://ballerina.io/learn/test-ballerina-code/mocking/

Reply:
Do you have a sample for this that allows maintaining concurrency safety?